flink-tests/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterStopWithSavepointIT.java

```diff
@@ -194,6 +197,30 @@ private void throwingExceptionOnCallbackWithRestartsHelper(final boolean termina
 		assertThat(getJobStatus(), either(equalTo(JobStatus.CANCELLING)).or(equalTo(JobStatus.CANCELED)));
 	}
 
+	@Test
+	public void testRestartCheckpointCoordinatorIfStopWithSavepointFails() throws Exception {
+		setUpJobGraph(ExceptionOnCallbackStreamTask.class, RestartStrategies.noRestart());
+
+		try {
+			Files.setPosixFilePermissions(savepointDirectory, Collections.emptySet());
+		} catch (IOException e) {
+			Assume.assumeNoException(e);
+		}
+
+		try {
+			stopWithSavepoint(true).get();
+			fail();
+		} catch (Exception e) {
+			assertThat(ExceptionUtils.findThrowable(e, CheckpointException.class).isPresent(), equalTo(true));
```

Comment:
1. I think you should be more specific and verify for example exact `CheckpointFailureReason`.
2. If the exception is not what you expected, you should just `throw e` it, instead of using `assertThat`, to get an even nicer exception message with a stack trace.